### DISCUSSION TITLE
Issue Report - TestSimpleExecuteAsync for SpotifyApiThreading

### DISCUSSION THREAD

**Discussion by Shenkeeee** [spotify-web-api-java/spotify-web-api-java#377] [General]
Hi,

I'd like to report an issue with the `executeAsync` method for the `SpotifyApiThreading` class The test, which aims to validate asynchronous execution using `CompletableFuture`, is currently failing for us. The expected outcome is the completion of `CompletableFuture` with a non-null result, but the test indicates otherwise. A revision might be requested to ensure the expected functionality of the `SpotifyApiThreading` class.

Thanks!
Mate

File path: `spotify-web-api-java/src/main/java/se/michaelthelin/spotify/SpotifyApiThreading.java`
Test custom file path: `spotify-web-api-java/src/test/java/se/michaelthelin/spotify/Ta23BravoTests.java` (only exists locally, mentioned for recreational purposes)

```Java
/**
 * Test: Testing simple asynchronous execution in the SpotifyApiThreading class.
 * Objective: Verifies that CompletableFuture works correctly in the executeAsync method.
 * Expected result: CompletableFuture completes and the result is not null.
 */
@Test
@DisplayName("Testing simple asynchronous execution in the SpotifyApiThreading class")
public void testSimpleExecuteAsync() {
  Callable<Integer> simpleCallable = () -> 42;

  CompletableFuture<Integer> future = SpotifyApiThreading.executeAsync(simpleCallable);

  assertTrue(future.isDone(), "CompletableFuture has completed.");

  assertNotNull(future.join(), "The result is not null.");
}
```

**Comment by dargmuesli** [comment]
Where exactly do you see the `TestSimpleExecuteAsync` method in this repository?

**Comment by Shenkeeee** [comment]
Let me clarify some of the descriptions.

Derived from the home repository, the path of the file is here, in the SpotifyApiThreading method:
spotify-web-api-java/src/main/java/se/michaelthelin/spotify/SpotifyApiThreading.java"

The main issue arises with a `CompletableFuture` that doesn't seem to complete under normal conditions. Despite being an asynchronous method, allowing other processes to initiate while it is running, it should have normally completed by the time it reaches the isDone method, based on the simplicity of the simpleCallable lambda method.

The isDone is a part of the `CompletableFuture` class, It should return true, if: 
"Returns true if completed in any fashion: normally, exceptionally, or via cancellation."

**Comment by Shenkeeee** [comment]
The test itself was created locally by us and was not originally included in the file system. Relatively, it can be found in our file system and potentially be recreated here:

spotify-web-api-java/src/test/java/se/michaelthelin/spotify/Ta23BravoTests.java

If needed, I can provide the whole file, the exact test that fails was provided in the original post.

**Reply by dargmuesli** [reply]
Ah, I see.

1. first of all, please use [code formatting](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting) in your original post so that I can read or copy the code properly.
2. for issues like the one you describe which include custom code, please create a fork of this repository (as you already seem to have done) and provide a failing test through that fork. You can show the failure e.g. by providing a link to the failing CI run or maybe even just the repository itself. In the best case that repository would only contain the minimal code that would make the issue clear.
3. also, you can comment directly under my answers instead of creating new sub-threads so this discussion looks a bit more orderly.
